aspnetcore/security/key-vault-configuration.md

```diff
@@ -0,0 +1,144 @@
+
+```csharp
+var store = new X509Store(StoreLocation.CurrentUser);
+store.Open(OpenFlags.ReadOnly);
+var cert = store.Certificates.Find(X509FindType.FindByThumbprint, config["CertificateThumbprint"], false);
+
+builder.AddAzureKeyVault(
+    config["Vault"],
+    config["ClientId"],
+    cert.OfType<X509Certificate2>().Single(),
+    new EnvironmentSecretManager(env.ApplicationName));
+store.Close();
+
+Configuration = builder.Build();
+```
+
+## Creating key vault secrets and loading configuration values
+1. Create a key vault and set up Azure Active Directory (Azure AD) for the application following the guidance in [Get started with Azure Key Vault](https://azure.microsoft.com/documentation/articles/key-vault-get-started/).
+  * Add secrets to the key vault using the Azure PowerShell Module, the Azure Management API, or the Azure Portal. Secrets are uploaded as either *Manual* or *Certificate* secrets. *Certificate* secrets are certificates for use by applications and services but are not supported by the configuration provider. You should use the *Manual* option to create name-value pair secrets for use with the configuration provider.
+    * Hierarchical values (configuration sections) use `--` (two dashes) as a separator.
+    * For the sample application, create two *Manual* secrets with the following name-value pairs:
+      * `MySecret`: `secret_value_1`
+      * `Section--MySecret`: `secret_value_2`
+  * Register the sample application with Azure Active Directory.
+  * Authorize the application to access the key vault. When you use the `Set-AzureRmKeyVaultAccessPolicy` PowerShell cmdlet to authorize the application to access the key vault, provide `List` and `Get` access to secrets with `-PermissionsToKeys list,get`.
+2. Update the application's *appsettings.json* file with the values of `Vault`, `ClientId`, and `ClientSecret`.
+3. Run the sample application, which obtains its configuration values from `IConfigurationRoot` with the same name as the secret name.
+  * Non-hierarchical Values: The value for `MySecret` is obtained with `config["MySecret"]`.
+  * Hierarchical Values (sections): Use `:` (colon) notation or the `.GetSection()` method.
+    * `config["Section:MySecret"]`
+    * `config.GetSection("Section")["MySecret"]`
+
+![Browser window showing secret values loaded via the Azure Key Vault Configuration Provider](key-vault-configuration/_static/sample-output.png)
+
+## Reloading secrets
+Secrets are cached until `IConfigurationRoot.Reload()` is called. Expired, disabled, and updated secrets in the key vault are not respected by the application until `Reload()` is executed.
+
+```csharp
+Configuration.Reload();
+```
+
+## Disabled and expired secrets
+Disabled and expired secrets throw a `KeyVaultClientException` at runtime. Therefore, it's important that you update your application or replace or update a disabled or expried secret in the key vault before the provider attempts to load or reload a disabled or expired secret.
```

Comment:
You could probably even go with
Disabled and expired secrets throw a `KeyVaultClientException` ~~at runtime~~. To prevent your app from throwing, replace your app or update the disabled/expired secret.